.github/scripts/crib/pr-comment-crib-env.js

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env node
+
+const core = require("@actions/core");
+const github = require("@actions/github");
+const { Octokit } = require("@octokit/rest");
+
+async function commentExists(octokit, owner, repo, prNumber, uniqueIdentifier) {
+  // This will automatically paginate through all comments
+  const comments = await octokit.paginate(octokit.rest.issues.listComments, {
+    owner,
+    repo,
+    issue_number: prNumber,
+  });
+
+  // Check each comment for the unique identifier
+  return comments.some((comment) => comment.body.includes(uniqueIdentifier));
+}
+
+async function run() {
+  try {
+    const token = process.env.GITHUB_TOKEN;
+    const octokit = new Octokit({ auth: token });
+
+    const context = github.context;
+    const labelsToCheck = ["crib"];
+    const { owner, repo } = context.repo;
+    const prNumber = context.issue.number;
+
+    if (!prNumber) {
+      core.setFailed("Could not get PR number from context");
+      return;
+    }
+
+    // Comment header and unique identifier
+    const commentHeader = "## CRIB Environment Details";
+
+    // Check if the comment already exists
+    if (await commentExists(octokit, owner, repo, prNumber, commentHeader)) {
+      core.info("CRIB environment comment already exists. Skipping.");
+      return;
+    }
+
+    // List labels on the PR
+    const { data: labels } = await octokit.rest.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number: prNumber,
+    });
+
+    // Check if any label matches the labelsToCheck
+    const labelMatches = labels.some((label) =>
```

Comment:
would it make sense to check for `crib` label first before checking for duplicate comment? presumably most PRs don't have the `crib` label?

Reply:
Great catch, updated order in 5d672779a43a888b5a2657dcd89503be509793da